src/aws-cpp-sdk-transfer/include/aws/transfer/TransferHandle.h

```diff
@@ -77,6 +77,8 @@ namespace Aws
                 bool IsLastPart() { return m_lastPart; }
                 void SetLastPart() { m_lastPart = true; }
 
+                Aws::String GetChecksum() const { return m_checksum; };
+                void SetChecksum(const Aws::String& checksum) { m_checksum = checksum; }
```

Comment:
should it really have a setter? what happens if you set it mid transfer?

Reply:
yeah fair

Reply:
This isnt on the public interface this is on "part state" that keeps track of one the parts being processed, we attach the checksum to the part so that when we call CompleteMultipart, we know the checksum of the part.